### ISSUE TITLE
Random Forest Inspired Neural Networks

### ISSUE THREAD

**Issue by lwill001** [Noahhobe/TeamOne2020#54]
This article proposes integrating a neural network with a random tree form. Neural networks do not perform as well in comparison to ML methods such as random forests. A carefully designed neural network with random forest structure can have better generalization ability. This architecture is more powerful than random forests, because the back-propagation algorithm that aids feedforward neural networks reduces to a more powerful and generalized way of constructing a decision tree. 
In tree-structured neural network for classification,  all the features are aggregated to form p, which is used as input to a softmax function (page 6, figure 7) to generate empirical class distributions for predicting the class label. For regression, instead of using the softmax to predict the class label, linear regression is used to predict the numerical target. https://dl.acm.org/doi/pdf/10.1145/3232230
